Source/BoxUtils/BoxPointers.hpp

```diff
@@ -28,26 +28,29 @@ class BoxPointers
                                               //! two values corresponding to
                                               //! adjacent coordinates in all
                                               //! directions.
+    const int m_in_num_comps;
 
     std::vector<double *> m_out_ptr;
     std::array<int, CH_SPACEDIM> m_out_stride; //!< Distance in memory between
                                                //! two values corresponding to
                                                //! adjacent coordinates in all
                                                //! directions.
+    const int m_out_num_comps;
```

Comment:
We could make all these member variables private/protected and then make `Cell` a friend class. Nothing else _should_ be accessing them anyway.

Reply:
On second thoughts, since also the `FourthOrderDerivatives` class relies on public access to these members, I think we should leave it as it is. I should point out that both `m_in_num_comps` and `m_out_num_comps` are `const` so they can't be modified by the user. Of course, in theory the user could accidentally modify the size of `m_in_ptr` and `m_out_ptr` but I think the risk of this happening is very low.

Reply:
I think `FourthOrderDerivatives` depends only on `m_in_ptr`  and `m_out_ptr` being accessible (and even these could be accessed with  getter functions). I was thinking about something like 866a542f886f2bb10f18463c1210caed6f2699dd.

Anyway I will accept the pull request  as it stands.